bundle/src/main/java/com/adobe/acs/commons/redirects/filter/RedirectFilter.java

```diff
@@ -156,7 +158,7 @@
             policy = ReferencePolicy.DYNAMIC,
             policyOption = ReferencePolicyOption.GREEDY
     )
-    private LocationHeaderAdjuster urlAdjuster;
+    private volatile LocationHeaderAdjuster urlAdjuster;

```

Comment:
usually a static policy means much less overhead, as those services should not change often....

Reply:
@YegorKozlov please push the same fix on vanityurlservice class for VanityUrlAdjuster referencce. @kwin can you update the bug description.